### ISSUE TITLE
R_IPI bool r_bin_file_set_cur_binfile_obj(RBin *bin, RBinFile *bf, RBinObject *obj) {

### ISSUE THREAD

**Issue by radare** [radare/radare2#12084]
Change signature to `R_IPI bool r_bin_file_set_cur_binfile_obj(RBin *bin, RBinFile *bf)`

because the last argument is implicit

**Comment by ret2libc** [comment]
Yes, I'll do it soon

**Comment by ret2libc** [comment]
actually... the last argument is necessary:
```
R_IPI bool r_bin_file_set_cur_binfile_obj(RBin *bin, RBinFile *bf, RBinObject *obj) {
	r_return_val_if_fail (bin && bf, false);
	if (!obj) {
		return false;
	}
	bin->file = bf->file;
	bin->cur = bf;
	bin->narch = bf->narch;
	bf->o = obj;
	RBinPlugin *plugin = r_bin_file_cur_plugin (bf);
	if (bin->minstrlen < 1) {
		bin->minstrlen = plugin? plugin->minstrlen: bin->minstrlen;
	}
	return true;
}
```

You cannot remove it, because this function is used to set `bf->o`, that means that `bf->o` is not set before. Can I close this?

**Comment by radare** [comment]
imho the function name is pretty bad, as well as the code inside.

* returns false if any of the args is null, so you cant unset a binobj.  then it should be in the val_if_fail
* sets some internal vars from the binfile as "global" in bin->, those are weak pointers which have no safetiness to break, and there must be a generic way to access those instead of "caching" them globally
* follows the concept of bin->cur which should be avoided
* plugin->minstrlen should have a valid value range, this is not checked
* r_bin_file_cur_plugin is pretty long and boring api to just do bf->curplugin (which also have a pretty bad name)

i mean, this code hurts me in several ways

**Comment by ret2libc** [comment]
Yeah I agree there are many things to fix, though they will probably be resolved once we start merging RBinFile/RBinObject together... I will start the merge soon, but not sure it will land for 3.2

**Comment by ret2libc** [comment]
I meant 3.1

**Comment by radare** [comment]
The function name is confusing and i dont think there must be any setter to fill this field because it must be done when calling the plugin

**Comment by radare** [comment]
i will focus on RBin and RAnal on next release. we shouldnt be delaying issues like this that much

**Comment by radare** [comment]
I did tonss of changes sin RBin related to this, actually i think the function hsould be renamed from `r_bin_file_set_cur_binfile_obj` to `r_bin_file_set_obj`. right onw theres only one rbinobj for eachh rbinfile, so the only problem is when this is null, becaues the plugin didnt initialized anything and there's some dupped info between binfile and binobj which may be deleted

**Comment by trufae** [comment]
done and resolved many years ago
